### DISCUSSION TITLE
GraphQL Error - 502 HTTP Code From Prisma Server -  [Closed connection]

### DISCUSSION THREAD

**Discussion by suhailga** [prisma/prisma#3155] [Q&A] [answered]
I am using `prisma1` with `mongo`. The performance is too bad. I have added indexes in Mongo but the overall query time is big. Also I have been getting this error in prisma logs

`    [INFO] Closed connection [connectionId{localValue:100, serverValue:505819}] to 122.42.142.32:27017 because there was a socket exception raised by this connection.
    [INFO] Closed connection [connectionId{localValue:9, serverValue:505727}] to 122.42.142.32:27017 because there was a socket exception raised on another connection from this pool.
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
        ... 3 more
    java.lang.OutOfMemoryError: Java heap space
    [INFO] Closed connection [connectionId{localValue:33, serverValue:505751}] to 122.42.142.32:27017 because there was a socket exception raised on another connection from this pool.
    [INFO] Closed connection [connectionId{localValue:35, serverValue:505754}] to 122.42.142.32:27017 because there was a socket exception raised on another connection from this pool.
    [INFO] Closed connection [connectionId{localValue:91, serverValue:505784}] to 122.42.142.32:27017 because there was a socket exception raised on another connection from this pool.
    [INFO] Closed connection [connectionId{localValue:32, serverValue:505750}] to 122.42.142.32:27017 because there was a socket exception raised on another connection from this pool.
    [INFO] Closed connection [connectionId{localValue:10, serverValue:505728}] to 122.42.142.32:27017 because there was a socket exception raised on another connection from this pool.
    [INFO] Closed connection [connectionId{localValue:22, serverValue:505739}] to 122.42.142.32:27017 because there was a socket exception raised on another connection from this pool.
    [INFO] Closed connection [connectionId{localValue:28, serverValue:505746}] to 122.42.142.32:27017 because there was a socket exception raised on another connection from this pool.
    [INFO] Closed connection [connectionId{localValue:95, serverValue:505816}] to 122.42.142.32:27017 because there was a socket exception raised on another connection from this pool.
    [INFO] Closed connection [connectionId{localValue:31, serverValue:505748}] to 122.42.142.32:27017 because there was a socket exception raised on another connection from this pool.
    [INFO] Closed connection [connectionId{localValue:43, serverValue:505761}] to 122.42.142.32:27017 because there was a socket exception raised on another connection from this pool.
    [INFO] Closed connection [connectionId{localValue:73, serverValue:505805}] to 122.42.142.32:27017 because there was a socket exception raised on another connection from this pool.
    [INFO] Closed connection [connectionId{localValue:6, serverValue:505725}] to 122.42.142.32:27017 because there was a socket exception raised on another connection from this pool.
    [INFO] Closed connection [connectionId{localValue:17, serverValue:505733}] to 122.42.142.32:27017 because there was a socket exception raised on another connection from this pool.
    [INFO] Closed connection [connectionId{localValue:51, serverValue:505769}] to 122.42.142.32:27017 because there was a socket exception raised on another connection from this pool.
    [INFO] Closed connection [connectionId{localValue:14, serverValue:505731}] to 122.42.142.32:27017 because there was a socket exception raised on another connection from this pool.`

Why does this occur? How do I fix this?

**Comment by janpio** [comment, marked as answer]
Duplicate of https://github.com/prisma/prisma1/issues/5102

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
